Sources/Datadog/Core/SessionTracer.swift

```diff
@@ -0,0 +1,54 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import _Datadog_Private
+
+// TODO: RUMM-300 This should probably be a part of DDTracer
+public extension Datadog {
+    internal typealias RequestInterceptor = (URLRequest) -> URLRequest
+    internal typealias TaskObserver = (URLSessionTask) -> Void
+
+    static func trace(_ session: URLSession) throws {
+        let requestInterceptor: RequestInterceptor = { originalRequest in
+            // TODO: RUMM-300 Set tracer HTTP header values
+            return originalRequest
+        }
+        let taskObserver: TaskObserver = { task in
+            let observation: NSKeyValueObservation
+            var oldState: URLSessionTask.State? = nil
+            // URLSessionTask.state does NOT support NSKeyValueObservedChange
+            observation = task.observe(\.state) { observedTask, _ in
+                let newState = observedTask.state
+                switch (oldState, newState) {
+                case (.some(.running), .running): // was running, is still running
+                    break
+                case (_, .running): // TODO: RUMM-300 Start span
```

Comment:
Among [all the possible `task` states](https://developer.apple.com/documentation/foundation/urlsessiontask/state#topics): `running`, `suspended`, `canceling`, `completed`, why `(_, .running)` for determining the "initial" one to start the span? Shouldn't this be `(.suspended, .running)` or `(nil, .running)`?

Reply:
i will probably remove this block before merging this PR as it cannot do anything now without `tracing` branch 🧹 

you are right, i think especially `stop` event is problematic here: what if we `downloadTask.suspend()`?
but i just left this naive way of catching start/stop events to showcase how it will be used in the future

Reply:
🎗️ Again, I left a `todo` for now, we can implement it when we plug `tracer`